File: e-shop-admin/core-ui/src/app/views/product-management/product-category/service/product-category.service.ts
import { Injectable, SkipSelf, Optional } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { mainURL } from '../../../Common/common-url-main';
import { CommonCustomResponce } from '../../../common/common-custom-responce';
import { ProductCategory } from '../model/product-category';

@Injectable({
    providedIn: 'root'
  })
export class ProductCategoryService{
    url : string = mainURL+"product-category";

    constructor(private _http: HttpClient){} //@SkipSelf()  


    save(mainModel : ProductCategory){
        return this._http.post<CommonCustomResponce>(this.url,mainModel);
    }

    delete(id: any) {
        return this._http.delete<CommonCustomResponce>(this.url+"/"+id);
      }
     
    getAll() {
        return this._http.get<ProductCategory[]>(this.url);
    }

    getById(id: any) {
        return this._http.get<ProductCategory>(this.url+"/"+id);
    }
      

}

File: e-shop-admin/ecart/src/main/java/com/ecart/searvice/ProductCategoryService.java
/**
 * 
 */
package com.ecart.searvice;

import java.util.List;

import com.ecart.domain.ProductCategory;
import com.ecart.model.CommonCustomResponce;

/**
 * @author Tharindu   Feb 19, 2020
 *
 */
public interface ProductCategoryService {

	/**
	 * @param productCategory
	 * @return
	 */
	CommonCustomResponce createUpdate(ProductCategory productCategory);

	/**
	 * @return
	 */
	List<ProductCategory> getAll();

	/**
	 * @param id
	 * @return
	 */
	ProductCategory getById(Long id);

	/**
	 * @param productCategory
	 * @return
	 */
	CommonCustomResponce delete(Long id);

}
